components/psst/browser/core/psst_rule_service.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_RULE_SERVICE_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_RULE_SERVICE_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/component_export.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/singleton.h"
+#include "base/memory/weak_ptr.h"
+#include "brave/components/psst/browser/core/psst_rule.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+class GURL;
+
+namespace psst {
+
+class COMPONENT_EXPORT(PSST_BROWSER_CORE) PsstRuleService {
+ public:
+  PsstRuleService(const PsstRuleService&) = delete;
+  PsstRuleService& operator=(const PsstRuleService&) = delete;
+  ~PsstRuleService();
+  COMPONENT_EXPORT(PSST_BROWSER_CORE)
+  void CheckIfMatch(const GURL& url,
+                    base::OnceCallback<void(MatchedRule)> cb) const;
+  COMPONENT_EXPORT(PSST_BROWSER_CORE)
+  static PsstRuleService* GetInstance();  // singleton
+  void LoadPsstRules(const base::FilePath& path);
+
+ protected:
+  PsstRuleService();
```

Comment:
why is this protected and not private? `DefaultSingletonTraits` is a friend so it can access private